source/content/guides/wordpress-configurations/07-wordpress-font-library.md

```diff
@@ -0,0 +1,78 @@
+innav: [false]
+categories: [config]
+cms: [wordpress]
+audience: [development]
+product: [--]
+integration: [plugins]
+tags: [workflow, code]
+contributors: [jazzsequence]
+permalink: docs/guides/wordpress-configurations/wordpress-font-library
+---
+
+This section provides information on how to use the WordPress Font Library on Pantheon.
+
+[WordPress 6.5 (release name)]() introduced a new [Font Library]() feature. The Font Library allows you to upload fonts to your WordPress site or install any of the fonts available from Google's font library. In anticipation of this, Pantheon has added a feature to our [Pantheon MU Plugin](https://github.com/pantheon-systems/pantheon-mu-plugin) to store those fonts in a writeable (`wp-content/uploads/fonts/`) directory, so that you can use the feature without any issues after updating your sites to 6.5. This changes the default WordPress core behavior of storing fonts in `wp-content/fonts/` to work on Pantheon.
+
+## Using the `font_dir` Filter
+
+You can change the directory where fonts are stored by using the WordPress core filter `font_dir` like this:
+
+```php
+add_filter( 'font_dir', function( $defaults ) {
+	$font_dir = '/path/to/your/custom/dir';
+	$font_url = site_url( $font_dir );
+	
+	$defaults['path'] = $font_dir;
+	$defaults['url'] = $font_url;
+	$defaults['basedir'] = $font_dir;
+	$defaults['baseurl'] = $font_url;
+	
+	return $defaults;
+} );
+```
+
+The default priority for WordPress filters is `10`. We have set _our_ `font_dir` filter to priority `9` so it allows you to override our modification.
+
+<Alert title="Note" type="info">
+
+You cannot use the function `wp_get_upload_dir()` inside the `font_dir` filter because it will cause an infinite loop. Instead, if you want to use the `wp_get_upload_dir()` function, you can use a global variable as [we do in our mu-plugin](https://github.com/pantheon-systems/pantheon-mu-plugin/blob/main/inc/fonts.php) or [remove the filter and then re-add it](https://github.com/WordPress/wordcamp.org/pull/1245/files#diff-e441f1053cefcd468bd20fed91d1aac5e902871d7c564be909fc35590f9c3082R635-R637).
+
+For more information, refer to [this Gutenberg issue](https://github.com/WordPress/gutenberg/issues/58696).
+
+</Alert>
+
+## Considerations
+
+WordPress handles fonts more like **media files** than **plugins or themes**. This means that when fonts are added to one Pantheon site environment, they will not necessarily exist in your other WordPress environments (e.g. installing a font on Dev does not mean it will push to Test when you deploy). In WordPress, fonts have two parts, there is an underlying font post type (similar to the `attachment` post type for media files) and the physical font files themselves. Without the font post type existing in the database, WordPress has no way of knowing that a font is installed (in the same way that WordPress has no way that a particular image exists in the `/uploads` directory if it was not uploaded via the media library).
```

Comment:
```suggestion
WordPress handles fonts more like **media files** than **plugins or themes**. This means that when fonts are added to one Pantheon site environment, they will not necessarily exist in your other environments (e.g. installing a font on Dev does not mean it will push to Test when you deploy). In WordPress, fonts have two parts, there is a font post type (similar to the `attachment` post type for media files) and the font files themselves. Without the font post type existing in the database, WordPress has no way of knowing that a font is installed (in the same way that WordPress has no way that a particular image exists in the `/uploads` directory if it was not uploaded via the media library).
```